app/Server.php

```diff
@@ -93,4 +127,181 @@ public function stats()
+                        // Meeting was created via a different system, ignore
+                        continue;
+                    }
+
+                    // Save current live room status and build archival data
+                    $meetingStat                            = new MeetingStat();
+                    $meeting->room->participant_count       = $meetingStat->participant_count       = $bbbMeeting->getParticipantCount();
+                    $meeting->room->listener_count          = $meetingStat->listener_count          = $bbbMeeting->getListenerCount();
+                    $meeting->room->voice_participant_count = $meetingStat->voice_participant_count = $bbbMeeting->getVoiceParticipantCount();
+                    $meeting->room->video_count             = $meetingStat->video_count             = $bbbMeeting->getVideoCount();
+
+                    if (setting('log_attendance')) {
+                        $attendees = [];
+                        foreach ($bbbMeeting->getAttendees() as $attendee) {
+                            array_push($attendees, $attendee->getFullName());
+                        }
+                        $meetingStat->attendees = json_encode($attendees);
+                    }
+
+                    $meeting->stats()->save($meetingStat);
+                    $meeting->room->save();
+                }
+
+                // Save current live server status
+                $this->participant_count       = $serverStat->participant_count;
+                $this->listener_count          = $serverStat->listener_count;
+                $this->voice_participant_count = $serverStat->voice_participant_count;
+                $this->video_count             = $serverStat->video_count;
+                $this->meeting_count           = $serverStat->meeting_count;
+                $this->status                  = ServerStatus::ONLINE;
+                $this->timestamps              = false;
+                $this->save();
+
+                $this->stats()->save($serverStat);
+            }
+        }
+        // find meetings that are marked as running in the database, but have not been found on the servers
+        // fix the end date in the database to current timestamp
+        $meetingsNotRunningOnServers = $allRunningMeetingsInDb->diff($allRunningMeetingsOnServers);
+        foreach ($meetingsNotRunningOnServers as $meetingId) {
+            $meeting = Meeting::find($meetingId);
+            if ($meeting != null && $meeting->end == null) {
+                $meeting->end = date('Y-m-d H:i:s');
+                $meeting->save();
+            }
+        }
+    }
+
+    /**
+     * Delete Server, only possible if no meetings from this system are running and the server is disabled
+     * @return bool|null
+     * @throws \Exception
+     */
+    public function delete()
+    {
+        if ($this->status != ServerStatus::DISABLED || $this->meetings()->whereNull('end')->count() != 0) {
```

Comment:
Please use a event for this like for the `updating` event!

Reply:
Changed